src/server/auth/cmds/cmds.go

```diff
@@ -130,21 +163,57 @@ Activate Pachyderm's auth system, and restrict access to existing data to the ro
 
 				if _, err := c.SetConfiguration(c.Ctx(),
 					&auth.SetConfigurationRequest{Configuration: &auth.OIDCConfig{
-						Issuer:          "http://localhost:30658/",
-						ClientID:        "pachd",
+						Issuer:          issuer,
+						ClientID:        clientId,
 						ClientSecret:    oidcClient.Client.Secret,
-						RedirectURI:     "http://localhost:30657/authorization-code/callback",
+						RedirectURI:     redirect,
 						LocalhostIssuer: true,
 					}}); err != nil {
 					return errors.Wrapf(grpcutil.ScrubGRPC(err), "failed to configure OIDC in pachd")
 				}
+			} else {
+				ec, err := newClient(true)
+				if err != nil {
+					return errors.Wrapf(grpcutil.ScrubGRPC(err), "failed to get enterprise server client")
+				}
+				idCfg, err := ec.GetIdentityServerConfig(c.Ctx(), &identity.GetIdentityServerConfigRequest{})
+				if err != nil {
+					return errors.Wrapf(grpcutil.ScrubGRPC(err), "failed to get identity server issuer")
+				}
+
+				oidcClient, err := ec.CreateOIDCClient(c.Ctx(), &identity.CreateOIDCClientRequest{
+					Client: &identity.OIDCClient{
+						Id:           clientId,
+						Name:         clientId,
+						TrustedPeers: trustedPeers,
+						RedirectUris: []string{redirect},
+					},
+				})
+				if err != nil {
+					return errors.Wrapf(grpcutil.ScrubGRPC(err), "failed to configure OIDC client ID")
+				}
+
+				if _, err := c.SetConfiguration(c.Ctx(),
+					&auth.SetConfigurationRequest{Configuration: &auth.OIDCConfig{
+						Issuer:          idCfg.Config.Issuer,
+						ClientID:        clientId,
+						ClientSecret:    oidcClient.Client.Secret,
+						RedirectURI:     redirect,
+						LocalhostIssuer: false,
+					}}); err != nil {
+					return errors.Wrapf(grpcutil.ScrubGRPC(err), "failed to configure OIDC in pachd")
+				}
 			}
 			return nil
 		}),
 	}
 	activate.PersistentFlags().BoolVar(&supplyRootToken, "supply-root-token", false, `
 Prompt the user to input a root token on stdin, rather than generating a random one.`[1:])
 	activate.PersistentFlags().BoolVar(&onlyActivate, "only-activate", false, "Activate auth without configuring the OIDC service")
+	activate.PersistentFlags().BoolVar(&enterprise, "enterprise", false, "Activate auth on the active enterprise context")
+	activate.PersistentFlags().StringVar(&issuer, "issuer", "http://localhost:30658/", "The issuer for the OIDC service")
```

Comment:
See my comment above. Is this just a URL or does the exact string matter later on?